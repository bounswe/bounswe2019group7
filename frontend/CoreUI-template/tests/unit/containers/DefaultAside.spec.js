import Vue from 'vue'
import { shallowMount } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue'
import DefaultAside from '@/containers/DefaultAside'

Vue.use(BootstrapVue)

describe('DefaultAside.vue', () => {
  it('has a name', () => {
    expect(DefaultAside.name).toMatch('DefaultAside')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(DefaultAside)
    expect(wrapper.element).toMatchSnapshot()
  })
})
