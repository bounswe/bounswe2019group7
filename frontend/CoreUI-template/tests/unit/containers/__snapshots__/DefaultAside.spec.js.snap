// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultAside.vue renders correctly 1`] = `
<b-tabs-stub
  tag="div"
>
  <b-tab-stub
    buttonid=""
    href="#"
    tag="div"
    title=""
  >
    <template>
      <i
        class="icon-list"
      />
    </template>
     
    <b-list-group-stub
      class="list-group-accent"
      tag="div"
    >
      <b-list-group-item-stub
        class="list-group-item-accent-secondary bg-light text-center font-weight-bold text-muted text-uppercase small"
        event="click"
        routertag="a"
        tag="div"
        target="_self"
      >
        
        Today
      
      </b-list-group-item-stub>
       
      <b-list-group-item-stub
        class="list-group-item-accent-warning list-group-item-divider"
        event="click"
        href="#"
        routertag="a"
        tag="div"
        target="_self"
      >
        <div
          class="avatar float-right"
        >
          <img
            alt="admin@bootstrapmaster.com"
            class="img-avatar"
            src="img/avatars/7.jpg"
          />
        </div>
         
        <div>
          Meeting with
          
          <strong>
            Lucas
          </strong>
        </div>
         
        <small
          class="text-muted mr-3"
        >
          <i
            class="icon-calendar"
          />
            1 - 3pm
        
        </small>
         
        <small
          class="text-muted"
        >
          <i
            class="icon-location-pin"
          />
            Palo Alto, CA
        
        </small>
      </b-list-group-item-stub>
       
      <b-list-group-item-stub
        class="list-group-item-accent-info"
        event="click"
        href="#"
        routertag="a"
        tag="div"
        target="_self"
      >
        <div
          class="avatar float-right"
        >
          <img
            alt="admin@bootstrapmaster.com"
            class="img-avatar"
            src="img/avatars/4.jpg"
          />
        </div>
         
        <div>
          Skype with 
          <strong>
            Megan
          </strong>
        </div>
         
        <small
          class="text-muted mr-3"
        >
          <i
            class="icon-calendar"
          />
            4 - 5pm
        </small>
         
        <small
          class="text-muted"
        >
          <i
            class="icon-social-skype"
          />
            On-line
        </small>
      </b-list-group-item-stub>
       
      <hr
        class="transparent mx-3 my-0"
      />
       
      <b-list-group-item-stub
        class="list-group-item-accent-secondary bg-light text-center font-weight-bold text-muted text-uppercase small"
        event="click"
        routertag="a"
        tag="div"
        target="_self"
      >
        
        Tomorrow
      
      </b-list-group-item-stub>
       
      <b-list-group-item-stub
        class="list-group-item-accent-danger list-group-item-divider"
        event="click"
        href="#"
        routertag="a"
        tag="div"
        target="_self"
      >
        <div>
          New UI Project - 
          <strong>
            deadline
          </strong>
        </div>
         
        <small
          class="text-muted mr-3"
        >
          <i
            class="icon-calendar"
          />
            10 - 11pm
        </small>
         
        <small
          class="text-muted"
        >
          <i
            class="icon-home"
          />
            creativeLabs HQ
        </small>
         
        <div
          class="avatars-stack mt-2"
        >
          <div
            class="avatar avatar-xs"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/2.jpg"
            />
          </div>
           
          <div
            class="avatar avatar-xs"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/3.jpg"
            />
          </div>
           
          <div
            class="avatar avatar-xs"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/4.jpg"
            />
          </div>
           
          <div
            class="avatar avatar-xs"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/5.jpg"
            />
          </div>
           
          <div
            class="avatar avatar-xs"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/6.jpg"
            />
          </div>
        </div>
      </b-list-group-item-stub>
       
      <b-list-group-item-stub
        class="list-group-item-accent-success list-group-item-divider"
        event="click"
        href="#"
        routertag="a"
        tag="div"
        target="_self"
      >
        <div>
          <strong>
            #10 Startups.Garden
          </strong>
           Meetup
        </div>
         
        <small
          class="text-muted mr-3"
        >
          <i
            class="icon-calendar"
          />
            1 - 3pm
        </small>
         
        <small
          class="text-muted"
        >
          <i
            class="icon-location-pin"
          />
            Palo Alto, CA
        </small>
      </b-list-group-item-stub>
       
      <b-list-group-item-stub
        class="list-group-item-accent-primary list-group-item-divider"
        event="click"
        href="#"
        routertag="a"
        tag="div"
        target="_self"
      >
        <div>
          <strong>
            Team meeting
          </strong>
        </div>
         
        <small
          class="text-muted mr-3"
        >
          <i
            class="icon-calendar"
          />
            4 - 6pm
        </small>
         
        <small
          class="text-muted"
        >
          <i
            class="icon-home"
          />
            creativeLabs HQ
        </small>
         
        <div
          class="avatars-stack mt-2"
        >
          <div
            class="avatar avatar-xs"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/2.jpg"
            />
          </div>
           
          <div
            class="avatar avatar-xs"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/3.jpg"
            />
          </div>
           
          <div
            class="avatar avatar-xs"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/4.jpg"
            />
          </div>
           
          <div
            class="avatar avatar-xs"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/5.jpg"
            />
          </div>
           
          <div
            class="avatar avatar-xs"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/6.jpg"
            />
          </div>
           
          <div
            class="avatar avatar-xs"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/7.jpg"
            />
          </div>
           
          <div
            class="avatar avatar-xs"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/8.jpg"
            />
          </div>
        </div>
      </b-list-group-item-stub>
    </b-list-group-stub>
  </b-tab-stub>
   
  <b-tab-stub
    buttonid=""
    href="#"
    tag="div"
    title=""
  >
    <template>
      <i
        class="icon-speech"
      />
    </template>
     
    <div
      class="p-3"
    >
      <div
        class="message"
      >
        <div
          class="py-3 pb-5 mr-3 float-left"
        >
          <div
            class="avatar"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/7.jpg"
            />
             
            <b-badge-stub
              class="avatar-status"
              event="click"
              routertag="a"
              tag="span"
              target="_self"
              variant="success"
            />
          </div>
        </div>
         
        <div>
          <small
            class="text-muted"
          >
            Lukasz Holeczek
          </small>
           
          <small
            class="text-muted float-right mt-1"
          >
            1:52 PM
          </small>
        </div>
         
        <div
          class="text-truncate font-weight-bold"
        >
          Lorem ipsum dolor sit amet
        </div>
         
        <small
          class="text-muted"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt...
        </small>
      </div>
       
      <hr />
       
      <div
        class="message"
      >
        <div
          class="py-3 pb-5 mr-3 float-left"
        >
          <div
            class="avatar"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/7.jpg"
            />
             
            <b-badge-stub
              class="avatar-status"
              event="click"
              routertag="a"
              tag="span"
              target="_self"
              variant="danger"
            />
          </div>
        </div>
         
        <div>
          <small
            class="text-muted"
          >
            Lukasz Holeczek
          </small>
           
          <small
            class="text-muted float-right mt-1"
          >
            1:52 PM
          </small>
        </div>
         
        <div
          class="text-truncate font-weight-bold"
        >
          Lorem ipsum dolor sit amet
        </div>
         
        <small
          class="text-muted"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt...
        </small>
      </div>
       
      <hr />
       
      <div
        class="message"
      >
        <div
          class="py-3 pb-5 mr-3 float-left"
        >
          <div
            class="avatar"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/7.jpg"
            />
             
            <b-badge-stub
              class="avatar-status"
              event="click"
              routertag="a"
              tag="span"
              target="_self"
              variant="info"
            />
          </div>
        </div>
         
        <div>
          <small
            class="text-muted"
          >
            Lukasz Holeczek
          </small>
           
          <small
            class="text-muted float-right mt-1"
          >
            1:52 PM
          </small>
        </div>
         
        <div
          class="text-truncate font-weight-bold"
        >
          Lorem ipsum dolor sit amet
        </div>
         
        <small
          class="text-muted"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt...
        </small>
      </div>
       
      <hr />
       
      <div
        class="message"
      >
        <div
          class="py-3 pb-5 mr-3 float-left"
        >
          <div
            class="avatar"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/7.jpg"
            />
             
            <b-badge-stub
              class="avatar-status"
              event="click"
              routertag="a"
              tag="span"
              target="_self"
              variant="warning"
            />
          </div>
        </div>
         
        <div>
          <small
            class="text-muted"
          >
            Lukasz Holeczek
          </small>
           
          <small
            class="text-muted float-right mt-1"
          >
            1:52 PM
          </small>
        </div>
         
        <div
          class="text-truncate font-weight-bold"
        >
          Lorem ipsum dolor sit amet
        </div>
         
        <small
          class="text-muted"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt...
        </small>
      </div>
       
      <hr />
       
      <div
        class="message"
      >
        <div
          class="py-3 pb-5 mr-3 float-left"
        >
          <div
            class="avatar"
          >
            <img
              alt="admin@bootstrapmaster.com"
              class="img-avatar"
              src="img/avatars/7.jpg"
            />
             
            <b-badge-stub
              class="avatar-status"
              event="click"
              routertag="a"
              tag="span"
              target="_self"
              variant="dark"
            />
          </div>
        </div>
         
        <div>
          <small
            class="text-muted"
          >
            Lukasz Holeczek
          </small>
           
          <small
            class="text-muted float-right mt-1"
          >
            1:52 PM
          </small>
        </div>
         
        <div
          class="text-truncate font-weight-bold"
        >
          Lorem ipsum dolor sit amet
        </div>
         
        <small
          class="text-muted"
        >
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt...
        </small>
      </div>
    </div>
  </b-tab-stub>
   
  <b-tab-stub
    buttonid=""
    href="#"
    tag="div"
    title=""
  >
    <template>
      <i
        class="icon-settings"
      />
    </template>
     
    <div
      class="p-3"
    >
      <h6>
        Settings
      </h6>
       
      <div
        class="aside-options"
      >
        <div
          class="clearfix mt-4"
        >
          <small>
            <b>
              Option 1
            </b>
          </small>
           
          <c-switch-stub
            checked="true"
            class="float-right"
            color="success"
            dataoff="Off"
            dataon="On"
            id="switch-id-8"
            label="true"
            size="sm"
            value="true"
            variant="pill"
          />
        </div>
         
        <div>
          <small
            class="text-muted"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </small>
        </div>
      </div>
       
      <div
        class="aside-options"
      >
        <div
          class="clearfix mt-3"
        >
          <small>
            <b>
              Option 2
            </b>
          </small>
           
          <c-switch-stub
            class="float-right"
            color="success"
            dataoff="Off"
            dataon="On"
            id="switch-id-9"
            label="true"
            size="sm"
            value="true"
            variant="pill"
          />
        </div>
         
        <div>
          <small
            class="text-muted"
          >
            Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </small>
        </div>
      </div>
       
      <div
        class="aside-options"
      >
        <div
          class="clearfix mt-3"
        >
          <small>
            <b>
              Option 3
            </b>
          </small>
           
          <c-switch-stub
            class="float-right"
            color="success"
            dataoff="Off"
            dataon="On"
            defaultchecked="true"
            disabled="true"
            id="switch-id-10"
            label="true"
            size="sm"
            value="true"
            variant="pill"
          />
        </div>
         
        <div>
          <small
            class="text-muted"
          >
            Disabled option.
          </small>
        </div>
      </div>
       
      <div
        class="aside-options"
      >
        <div
          class="clearfix mt-3"
        >
          <small>
            <b>
              Option 4
            </b>
          </small>
           
          <c-switch-stub
            checked="true"
            class="float-right"
            color="success"
            dataoff="Off"
            dataon="On"
            id="switch-id-11"
            label="true"
            size="sm"
            value="true"
            variant="pill"
          />
        </div>
      </div>
       
      <hr />
       
      <h6>
        System Utilization
      </h6>
       
      <div
        class="text-uppercase mb-1 mt-4"
      >
        <small>
          <b>
            CPU Usage
          </b>
        </small>
      </div>
       
      <b-progress-stub
        class="progress-xs"
        height="{}"
        max="100"
        precision="0"
        value="25"
        variant="info"
      />
       
      <small
        class="text-muted"
      >
        348 Processes. 1/4 Cores.
      </small>
       
      <div
        class="text-uppercase mb-1 mt-2"
      >
        <small>
          <b>
            Memory Usage
          </b>
        </small>
      </div>
       
      <b-progress-stub
        class="progress-xs"
        height="{}"
        max="100"
        precision="0"
        value="70"
        variant="warning"
      />
       
      <small
        class="text-muted"
      >
        11444GB/16384MB
      </small>
       
      <div
        class="text-uppercase mb-1 mt-2"
      >
        <small>
          <b>
            SSD 1 Usage
          </b>
        </small>
      </div>
       
      <b-progress-stub
        class="progress-xs"
        height="{}"
        max="100"
        precision="0"
        value="95"
        variant="danger"
      />
       
      <small
        class="text-muted"
      >
        243GB/256GB
      </small>
       
      <div
        class="text-uppercase mb-1 mt-2"
      >
        <small>
          <b>
            SSD 2 Usage
          </b>
        </small>
      </div>
       
      <b-progress-stub
        class="progress-xs"
        height="{}"
        max="100"
        precision="0"
        value="10"
        variant="success"
      />
       
      <small
        class="text-muted"
      >
        25GB/256GB
      </small>
    </div>
  </b-tab-stub>
</b-tabs-stub>
`;
